src/features/auth/whoami.ts

```diff
@@ -3,12 +3,58 @@ import type { RequestHandler } from 'express';
 import { sbvrUtils, permissions } from '@balena/pinejs';
 
 import { getUser } from '../../infra/auth/auth';
-import { captureException, handleHttpErrors } from '../../infra/error-handling';
+import {
+	captureException,
+	handleHttpErrors,
+	ThisShouldNeverHappenError,
+} from '../../infra/error-handling';
 
-import type { User } from '../../balena-model';
+import type { User, Application, Device } from '../../balena-model';
+import { UnauthorizedError } from '@balena/pinejs/out/sbvr-api/errors';
 
 const { api } = sbvrUtils;
 
+type ExpandedActor =
+	| {
+			id: number;
+			is_of__user: [Pick<User, 'id' | 'username' | 'email'>];
+			is_of__application: [];
+			is_of__device: [];
+	  }
+	| {
+			id: number;
+			is_of__user: [];
+			is_of__application: [Pick<Application, 'id' | 'slug'>];
+			is_of__device: [];
+	  }
+	| {
+			id: number;
+			is_of__user: [];
+			is_of__application: [];
+			is_of__device: [Pick<Device, 'id' | 'uuid'>];
+	  };
+
+type ActorResponse =
+	| {
+			id: number;
+			actorType: 'user';
+			actorTypeId: number;
+			username: string;
+			email: string | null;
+	  }
+	| {
+			id: number;
+			actorType: 'application';
+			actorTypeId: number;
+			slug: string;
+	  }
+	| {
+			id: number;
+			actorType: 'device';
+			actorTypeId: number;
+			uuid: string;
+	  };
+
```

Comment:
This interface is the only thing that bothers me a bit in the sense that the ` /user/v1/whoami` endpoint also has an "id" field and this one has an "id" as well, but they will differ as one is the user id and other the actor id. Maybe change the interface to explicitly be `actorId` instead of `id`? Any opinions are welcomed